tpu_inference/kernels/ragged_paged_attention/v3/tuned_block_sizes_hd64.py

```diff
@@ -288,17 +432,35 @@ def get_tuned_block_sizes(
         actual_num_kv_heads,
         head_dim,
         page_size * pages_per_seq,
+        sliding_window,
     )
-    device, page_size, dtypes, head_dims, max_model_len = keys
+    device, page_size, dtypes, head_dims, extra = keys
 
     try:
         bkv_p, bq = TUNED_BLOCK_SIZES[device][page_size][dtypes][head_dims][
-            max_model_len]
+            extra]
+        print('RPA v3 kernel: Found tuned sizes for %s', keys)
     except KeyError:
         logger.warning_once(
             'Couldn`t find tuned sizes for the RPA v3 kernel with %s', keys)
+        # When not available use a sensible default based on TPU version
+        # Set default block sizes for each tpu_version.
+        tpu_version = get_tpu_version()
+        if tpu_version < 4:
+            raise NotImplementedError('TPU version must be 4 or higher.')
+        match tpu_version:
+            case 4:
+                # TPUv4 has much smaller VMEM size so we pick fixed block sizes.
+                bkv_p, bq = (512 // page_size, 32)
+            case 7:
+                bkv_p, bq = (4096 // page_size, 32)
+            case _:
+                bkv_p, bq = (2048 // page_size, 32)
 
-    return (min(pages_per_seq, bkv_p), min(max_num_tokens, bq))
+        bkv_p, bq = (min(pages_per_seq, bkv_p), min(max_num_tokens, bq))
+
+    print('RPA v3 kernel tuned block sizes: bkv_p=%s, bq=%s', bkv_p, bq)
```

Comment:
Can you change this into logging info once?

Reply:
done